src/components/BecomeAProvider.svelte

```diff
@@ -0,0 +1,66 @@
+<script lang="ts">
+  import { dotApi } from '$lib/stores';
+  import { onMount } from 'svelte';
+  import type { web3Enable, web3FromSource } from '@polkadot/extension-dapp';
+  import { defaultDotApi } from '$lib/storeTypes';
+  import type { DotApi } from '$lib/storeTypes';
+  import { user } from '$lib/stores/userStore';
+  import { nonProviderAccountsStore } from '$lib/stores/accountsStore';
+  import BlockSection from './BlockSection.svelte';
+  import SelectNetworkAndAccount from './SelectNetworkAndAccount.svelte';
+  import CreateMsa from './CreateMsa.svelte';
+  import CreateProvider from './CreateProvider.svelte';
+  import EmailProviderRequest from './EmailProviderRequest.svelte';
+  import { pageContent } from '$lib/stores/pageContentStore';
+
+  let localDotApi: DotApi = defaultDotApi;
+  let thisWeb3FromSource: typeof web3FromSource;
+  let thisWeb3Enable: typeof web3Enable;
+
+  // a callback for when the user cancels this action
+  export let cancelAction = () => {
+    pageContent.login();
+  };
+
+  onMount(async () => {
+    const extension = await import('@polkadot/extension-dapp');
+    thisWeb3FromSource = extension.web3FromSource;
+    thisWeb3Enable = extension.web3Enable;
+  });
+
+  dotApi.subscribe((api) => (localDotApi = api));
+</script>
+
+<div id="become-a-provider" class="content-block flex w-single-block flex-col gap-4">
+  <BlockSection title="Become a provider">
+    <form class="flex w-[320px] flex-col gap-4">
+      <SelectNetworkAndAccount
+        selectedNetwork={$user.network}
+        accountsStore={$nonProviderAccountsStore}
+        accountSelectorTitle="Wallet Address"
+        accountSelectorPlaceholder="Select a wallet address"
+        noAccountsFoundErrorMsg="No accounts found.  Add an account to your wallet."
+      ></SelectNetworkAndAccount>
+      {#if $user.network != null && $user.network.name == 'MAINNET'}
+        <EmailProviderRequest />
+      {:else if $user.address !== ''}
+        {#if $user.msaId === 0}
+          <CreateMsa />
+        {:else}
+          <CreateProvider />
+        {/if}
+      {:else}
+        <button on:click|preventDefault={cancelAction} class="btn-no-fill text-left">Cancel</button>
+      {/if}
```

Comment:
I would consider moving this into it's own component in an effort to make BecomeAProvider as 'dumb' as possible.

Reply:
I don't see why.